development/gcp/cloud-functions/rotateserviceaccount/rotateserviceaccount.go

```diff
@@ -0,0 +1,125 @@
+package rotateserviceaccount
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/kyma-project/test-infra/development/gcp/pkg/cloudfunctions"
+	"github.com/kyma-project/test-infra/development/gcp/pkg/pubsub"
+	"github.com/kyma-project/test-infra/development/gcp/pkg/secretmanager"
+	"google.golang.org/api/iam/v1"
+)
+
+var (
+	projectID             string
+	secretManagerService  *secretmanager.Service
+	serviceAccountService *iam.Service
+)
+
+// ServiceAccountJSON stores Service Account athentication data
+type ServiceAccountJSON struct {
+	Type             string `json:"type"`
+	ProjectID        string `json:"project_id"`
+	PrivatekayID     string `json:"private_key_id"`
+	PrivateKey       string `json:"private_key"`
+	ClientEmail      string `json:"client_email"`
+	ClientID         string `json:"client_id"`
+	AuthURL          string `json:"auth_uri"`
+	TokenURI         string `json:"token_uri"`
+	AuthProviderCert string `json:"auth_provider_x509_cert_url"`
+	ClientCert       string `json:"client_x509_cert_url"`
+}
+
+func init() {
+	var err error
+	ctx := context.Background()
+
+	projectID, err = metadata.ProjectID()
+	if err != nil {
+		panic("failed to retrieve GCP Project ID, error: " + err.Error())
+	}
+
+	secretManagerService, err = secretmanager.NewService(ctx)
+	if err != nil {
+		panic("failed creating Secret Manager client, error: " + err.Error())
+	}
+
+	serviceAccountService, err = iam.NewService(ctx)
+	if err != nil {
+		panic("failed creating IAM client, error: " + err.Error())
+	}
+}
+
+// RotateServiceAccount adds new secret version in Secret Manager on pubsub rotate message
+func RotateServiceAccount(ctx context.Context, m pubsub.MessagePayload) error {
+	var err error
+	var secretRotateMessage pubsub.SecretRotateMessage
+	var secretData ServiceAccountJSON
+
+	// Create logger to use google cloud functions structured logging
+	logger := cloudfunctions.NewLogger()
+	// Set component for log entries to identify all messages for this function.
+	logger.WithComponent("kyma.prow.cloud-function.RotateServiceAccount")
+	// Set trace value for log entries to identify messages from one function call.
+	logger.GenerateTraceValue(projectID, "RotateServiceAccount")
+
+	if m.Attributes["eventType"] != "SECRET_ROTATE" {
+		logger.LogDebug("Unsupported event type: %s, quitting", m.Attributes["eventType"])
```

Comment:
You can filter pubsub messages by attribute value on pubsub subscription. This way function will not be called for eventType other than SECRET_ROTATE. If you will filter this on subscription, this message can be logged with severity ERROR.
```suggestion
		logger.LogError("Unsupported event type: %s, quitting", m.Attributes["eventType"])
```

Reply:
I'd have to change function trigger to HTTP